tools/CACertificates/README.md

```diff
@@ -0,0 +1,108 @@
+# Managing test CA certificates for samples and tutorials
+
+## WARNING
+Certificates created by these scripts **MUST NOT** be used for production.  They contain hard-coded passwords ("1234"), expire by default after 30 days, and most importantly are provided for demonstration purposes to help you quickly understand CA Certificates.  When productizing against CA Certificates, you'll need to use your own security best practices for certification creation and lifetime management.
+
+## Introduction
+This document helps create certificates for use in **pre-testing** IoT SDK's against the IoT Hub and Edge runtime.  In particular, the tools in this directory can be used to either setup CA Certificates (along with proof of possession) or Edge device certificates.  This document assumes you have basic familiarity with the scenario you are setting up for as well as some knowledge of PowerShell or Bash.
+
+This directory contains a PowerShell (PS1) and Bash script to help create **test** certificates for Azure IoT Hub's CA Certificate / proof-of-possession and/or Edge certificates.
+
+The PS1 and Bash scripts are functionally equivalent; they are both provided depending on your preference for Windows or Linux respectively.
+
+A more detailed document showing UI screen shots for CA Certificates and proof of possession flow is available from [the official documentation].
+
+A more detailed document explaining Edge and showing its use of certificates generated here is available from the [Edge gateway creation documentation].
+
+## USE
+
+## Step 1 - Initial Setup
+You'll need to do some initial setup prior to running these scripts.
+
+###  **PowerShell**
+* Get OpenSSL for Windows.
+  * See https://docs.microsoft.com/en-us/azure/iot-edge/how-to-create-transparent-gateway#install-openssl.
+* Start PowerShell as an Administrator.
+* `cd` to a working directory you want to run in.  All files will be created in this directory.
+* `cp *.cnf` and `cp ca-certs.ps1` from the directory this .MD file is located into your working directory.
+* Run `Set-ExecutionPolicy -ExecutionPolicy Unrestricted`.  You'll need this for PowerShell to allow you to run the scripts.
+* Run `. .\ca-certs.ps1` .  This is called dot-sourcing and brings the functions of the script into PowerShell's global namespace.
+* Run `Test-CACertsPrerequisites` to make sure all the required pre requisites are met and follow on the on screen prompts to resolve.
+
+###  **Bash**
+* Start Bash.
+* `cd` to the directory you want to run in.  All files will be created as children of this directory.
+* `cp *.cnf` and `cp certGen.sh` from the directory this .MD file is located into your working directory.
+* `chmod +x certGen.sh`
```

Comment:
You should be able to check this file in with execute mode. Then this step wouldn't be needed on linux.